src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+import java.util.Locale;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Checks for empty line separators at the beginning/end of a code block.
+ *
+ * <p> The policy to verify is specified using the {@link BlockWrapOption} class and
+ * defaults to {@link BlockWrapOption#EMPTY_LINE_ALLOWED}. Policies for the beginning and
+ * the end of the block are configured separately.
+ * </p>
+ *
+ * <p> The check is for any statement that have a body:
+ * {@link TokenTypes#ANNOTATION_ARRAY_INIT ANNOTATION_ARRAY_INIT},
+ * {@link TokenTypes#ANNOTATION_DEF ANNOTATION_DEF},
+ * {@link TokenTypes#ARRAY_INIT ARRAY_INIT},
+ * {@link TokenTypes#CASE_GROUP CASE_GROUP},
+ * {@link TokenTypes#CLASS_DEF CLASS_DEF},
+ * {@link TokenTypes#CTOR_DEF CTOR_DEF},
+ * {@link TokenTypes#ENUM_CONSTANT_DEF ENUM_CONSTANT_DEF},
+ * {@link TokenTypes#ENUM_DEF ENUM_DEF},
+ * {@link TokenTypes#INSTANCE_INIT INSTANCE_INIT},
+ * {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF},
+ * {@link TokenTypes#LAMBDA LAMBDA},
+ * {@link TokenTypes#LITERAL_CASE LITERAL_CASE},
+ * {@link TokenTypes#LITERAL_CATCH LITERAL_CATCH},
+ * {@link TokenTypes#LITERAL_DEFAULT LITERAL_DEFAULT},
+ * {@link TokenTypes#LITERAL_DO LITERAL_DO},
+ * {@link TokenTypes#LITERAL_ELSE LITERAL_ELSE},
+ * {@link TokenTypes#LITERAL_FINALLY LITERAL_FINALLY},
+ * {@link TokenTypes#LITERAL_FOR LITERAL_FOR},
+ * {@link TokenTypes#LITERAL_IF LITERAL_IF},
+ * {@link TokenTypes#LITERAL_NEW LITERAL_NEW},
+ * {@link TokenTypes#LITERAL_SWITCH LITERAL_SWITCH},
+ * {@link TokenTypes#LITERAL_SYNCHRONIZED LITERAL_SYNCHRONIZED},
+ * {@link TokenTypes#LITERAL_TRY LITERAL_TRY},
+ * {@link TokenTypes#LITERAL_WHILE LITERAL_WHILE},
+ * {@link TokenTypes#METHOD_DEF METHOD_DEF},
+ * {@link TokenTypes#OBJBLOCK OBJBLOCK},
+ * {@link TokenTypes#SLIST SLIST},
+ * {@link TokenTypes#STATIC_INIT STATIC_INIT}.
+ * </p>
+ *
+ * <p> By default the check will check the following statements:
```

Comment:
I think default option should be to enforce top empty line , it is same as http://checkstyle.sourceforge.net/config_whitespace.html#EmptyLineSeparator , and we need to make Check doing smth in code in other case user will be in illusion that activated check just did not found any violation.

Reply:
Indeed. The default options must enforce some style. I delayed this issue till we know what options should be by default.

Reply:
> I think default option should be to enforce top empty line

@romani By default, I am fine with the check doing no validation. If we specify values for top/bottom, then we always have to clear those values out whenever we want to add a new instance. If we keep them by default to `NONE`, then you don't need to always define a value over it if you want to keep it as `NONE.

> we need to make Check doing smth in code in other case user will be in illusion that activated check just did not found any violation

I believe we have some checks that actually do nothing unless configured otherwise.

Reply:
> whenever we want to add a new instance. 

but having module that do nothing could cause that illusion that it works and code comply. User is investigating configuration only when he have problems with validations that annoy. Use activate check, it give him bunch of violations, user do not like violation, user change Check.

> I believe we have some checks that actually do nothing unless configured otherwise.

yes, but only we completely not sure what to check, like illegal text, .....